framework/src/media/StreamBuffer.cpp

```diff
@@ -130,6 +141,47 @@ void StreamBuffer::notifyObserver(State st, ...)
 	}
 }
 
+StreamBuffer::Builder::Builder()
+	: mBufferSize(CONFIG_STREAM_BUFFER_SIZE_DEFAULT), mThreshold(CONFIG_STREAM_BUFFER_THRESHOLD_DEFAULT)
+{
+}
+
+StreamBuffer::Builder::~Builder()
+{
+}
+
+StreamBuffer::Builder &StreamBuffer::Builder::setBufferSize(size_t bufferSize)
+{
+	mBufferSize = bufferSize;
+}
+
+StreamBuffer::Builder &StreamBuffer::Builder::setThreshold(size_t threshold)
+{
+	mThreshold = threshold;
+}
+
+std::shared_ptr<StreamBuffer> StreamBuffer::Builder::build()
+{
+	if (mBufferSize == 0) {
+		mBufferSize = CONFIG_STREAM_BUFFER_SIZE_DEFAULT;
+	}
+
+	if (mThreshold == 0) {
+		mThreshold = CONFIG_STREAM_BUFFER_THRESHOLD_DEFAULT;
+	}
+
+	if (mThreshold > mBufferSize) {
+		mThreshold = mBufferSize;
+	}
+
+	auto instance = std::make_shared<StreamBuffer>(mBufferSize, mThreshold);
+	if (instance->init(mBufferSize)) {
+		return instance;
+	}
+
+	meddbg("init failed!\n", mBufferSize, mThreshold);
```

Comment:
meddbg("init failed! mBufferSize: %u, mThreshold: %u\n", mBufferSize, mThreshold);

Reply:
Fixed it